ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -374,8 +387,14 @@ private void registerDdlCommandTasks() {
       QueryMetadata queryMetadata = ksqlEngine.getQueryExecutionPlan(((CreateStreamAsSelect) statement).getQuery());
       if (queryMetadata.getDataSourceType() == DataSource.DataSourceType.KTABLE) {
         throw new KsqlException("Invalid result type. Your SELECT query produces a TABLE. Please "
-                + "use CREATE TABLE AS SELECT statement instead.");
+                                + "use CREATE TABLE AS SELECT statement instead.");
+      }
+      if (queryMetadata instanceof PersistentQueryMetadata) {
+        new AvroUtil().validatePersistentQueryResults((PersistentQueryMetadata) queryMetadata,
```

Comment:
Seems like it would be better for `AvroUtil.validatePersistentQueryResults` to be static?